canonicalwebteam/blog/common_view_logic.py

```diff
@@ -20,46 +26,48 @@ def get_index_context(page_param, articles, total_pages):
 
         category_ids = article["categories"]
 
+        # Can these calls be bundled?
+        first_item = True
         for category_id in category_ids:
             if category_id not in category_cache:
-                category_cache[category_id] = {}
-
+                try:
+                    resolved_category = api.get_category_by_id(category_id)
+                except Exception:
+                    resolved_category = None
+                category_cache[category_id] = resolved_category
+            if first_item:
+                article["display_category"] = category_cache[category_id]
+                first_item = False
+        first_item = True
         for group_id in article["group"]:
             if group_id not in group_cache:
-                group_cache[group_id] = {}
+                try:
+                    resolved_group = api.get_group_by_id(group_id)
+                except Exception:
+                    resolved_group = None
+                group_cache[group_id] = resolved_group
+            if first_item:
+                article["group"] = group_cache[group_id]
```

Comment:
Why is the first item special?

Reply:
This will be the group that gets displayed on the top of the article. It also determines the styling.
Since some articles have multiple groups, I have chosen to just pick the first one that is in the array of returned groups.

![image](https://user-images.githubusercontent.com/2843450/55314693-97b10600-5462-11e9-9319-afb0e30d6e8c.png)